python/ambassador/envoy/v3/v3listener.py

```diff
@@ -990,13 +1045,25 @@ def finalize_http(self) -> None:
             self._filter_chains.append(filter_chain)
 
     def as_dict(self) -> dict:
-        listener = {
+        listener: dict = {
             "name": self.name,
             "address": self.address,
+            "udp_listener_config": {},
             "filter_chains": self._filter_chains,
             "traffic_direction": self.traffic_direction
         }
 
+        if self.reuse_port == True:
+            listener['enable_reuse_port'] = self.reuse_port
```

Comment:
It looks like you're setting this if both TCP and UDP are set to the same port number.  That isn't necessary; TCP port numbers and UDP port numbers are totally separate in the Linux kernel; no need to set SO_REUSEPORT to listen on both TCP port X and UDP port X.

(Just to be sure, I verified this: I launched `socat TCP-LISTEN:8080,fork STDIO` in one terminal, `socat UDP-LISTEN:8080,fork STDIO` in another, and verified in a 3rd terminal that I could talk to each.)

Reply:
See Envoy's docs on what SO_REUSEPORT does on different platforms: https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/listener/v3/listener.proto#envoy-v3-api-field-config-listener-v3-listener-enable-reuse-port

Also, Envoy defaults it to "true", so the whole thing's double-pointless, I think.

Reply:
interesting...I see the docs say that and agree. I will double check but I thought the reason I added it explicitly was because it was crashing...I will try it with out and see.

Reply:
So, I tested and it worked so not sure where I got the idea that it needed to be set to make sure it worked. 🤷  I have removed it, built a test image and it came up with no problems in a cluster so we should be good without explicitly setting it.